client/landing/stepper/declarative-flow/internals/steps-repository/design-setup/unified-design-picker.tsx

```diff
@@ -953,4 +948,64 @@ function getBusinessPlanByTerm( term: string ) {
 	}
 }
 
+function getRequiredPlan( selectedDesign: Design | undefined, currentPlanSlug: string ) {
+	if ( ! selectedDesign || ! selectedDesign?.design_tier ) {
+		return;
+	}
+	// Different designs require different plans to unlock them, additionally the terms required can vary.
+	// A site with a plan of a given length cannot upgrade a plan of a shorter length. For example,
+	// if a site is on a 2 year starter plan and want to buy an explorer theme, they must buy a 2 year explorer plan
+	// not a 1 year explorer plan.
+	const tierMinimumUpsellPlan =
+		THEME_TIERS[ selectedDesign.design_tier as keyof typeof THEME_TIERS ]?.minimumUpsellPlan;
+
+	let requiredTerm;
+	if ( ! currentPlanSlug || isFreePlan( currentPlanSlug ) ) {
+		// Marketplace themes require upgrading to a monthly business plan or higher, everything else requires an annual plan.
+		if ( selectedDesign?.is_externally_managed ) {
+			requiredTerm = TERM_MONTHLY;
+		} else {
+			requiredTerm = TERM_ANNUALLY;
+		}
+	} else {
+		requiredTerm = getPlan( currentPlanSlug )?.term || '';
+	}
+
+	return getPlanByTerm( tierMinimumUpsellPlan, requiredTerm );
+}
+
+function getPlanByTerm( plan: string, term: string ) {
+	// This is icky, replace later with less hard-coding.
```

Comment:
I got rabbit-holed when addressing this before while working on a different approach. It'd be nice if there was a better way to approach it but it's not a blocker.

Reply:
Had a look around now I'm out of the hole and it's working, and this is now addressed much more nicely.